infra/cifuzz/actions/entrypoint.py

```
@@ -12,46 +12,69 @@
+  workspace = os.environ.get('GITHUB_WORKSPACE')
+  if not workspace:
+    logging.error('This script needs to be run in the Github action context.')
+    return 1
+  git_workspace = os.path.join(workspace, 'storage')
+  os.makedirs(git_workspace, exist_ok=True)
+  out_dir = os.path.join(workspace, 'out')
+  os.makedirs(out_dir, exist_ok=True)
 
   # Build the specified project's fuzzers from the current repo state.
-  print('Building fuzzers\nproject: {0}\nrepo name: {1}\ncommit: {2}'.format(
-      project_name, repo_name, commit_sha))
-  command = [
-      'python3', '/src/oss-fuzz/infra/cifuzz.py', 'build_fuzzers', project_name,
-      repo_name, commit_sha
-  ]
-  print('Running command: "{0}"'.format(' '.join(command)))
-  try:
-    subprocess.check_call(command)
-  except subprocess.CalledProcessError as err:
-    sys.stderr.write('Error building fuzzers: "{0}"'.format(str(err)))
-    return err.returncode
+  if not cifuzz.build_fuzzers(oss_fuzz_project_name, github_repo_name,
+                              commit_sha, git_workspace, out_dir):
+    logging.error('Error building fuzzers for project %s.',
+                  oss_fuzz_project_name)
+    return 1
 
   # Run the specified project's fuzzers from the build.
-  command = [
-      'python3', '/src/oss-fuzz/infra/cifuzz.py', 'run_fuzzers', project_name
-  ]
-  print('Running command: "{0}"'.format(' '.join(command)))
-  try:
-    subprocess.check_call(command)
-  except subprocess.CalledProcessError as err:
-    sys.stderr.write('Error running fuzzers: "{0}"'.format(str(err)))
-    return err.returncode
-  print('Fuzzers ran successfully.')
+  run_status, bug_found = cifuzz.run_fuzzers(oss_fuzz_project_name,
+                                             fuzz_seconds, out_dir)
+  if not run_status:
+    logging.error('Error occured while running fuzzers for project %s.',
+                  oss_fuzz_project_name)
+    return 1
+  if bug_found:
+    logging.debug('Bug found.')
+    return 2
```

Do we need to return an error code that isn't 0 or 1 here? Why not return 1? I can understand that we might need to return something else, just please explain why (even a comment explaining this would be good.

---

I will add a comment, but there are two types of crashes. One type is from an error in the CI code and the other is the detection of a bug. I wanted to distinguish between the two with error codes.